src/features/posts/postsSlice.ts

```diff
@@ -0,0 +1,56 @@
+/* eslint-disable no-param-reassign */
+
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+import { getUserPosts } from '../../api/posts';
+
+export const fetchUserPosts = createAsyncThunk<Post[], number>(
+  'posts/fetchUserPosts',
+  userId => {
+    const response = getUserPosts(userId);
+
+    return response;
+  },
+);
+
+const initialState: {
+  posts: Post[];
+  isLoading: boolean;
+  hasError: boolean;
+} = {
+  posts: [],
+  isLoading: false,
+  hasError: false,
+};
+
+export const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    setPosts: (state, action: PayloadAction<Post[]>) => {
+      state.posts = action.payload;
+    },
+  },
+  extraReducers: builder => {
```

Comment:
It's good practice to handle the `rejected` state in the `extraReducers` by storing the error in the state. This way, you can display an error message to the user or take other appropriate actions.